molgenis/molgenis-emx2
# feat: analytics service

Allow (admin) users setup analytics events. When the end user user the web interface and executes a certain action ( for example clicks on the 'contact' button) an signal is send to a analytics service ( third party; site-improve, google, .. or emx2 analytics endpoint). The app maintainer (admin ? )should be able to configure which events are send in response to what ui element the user interacts with ( initial 'click' event only). The maintainer should be able to configure these events on a per schema basis. 

What are the main changes you did:
- Add 'Tigger' model to the System schema ( a trigger hold the configuration as defined by the maintainer )
- Create a client module that fetches the triggers configured for a schema and initialises event triggers in the client application ( if necessary it fetches code to handle the triggers, i.e. send the analytics events for a certain ( set of ) provider(s).
- Create a API to c.r.u.d triggers
- Create a ui to  c.r.u.d triggers

how to test:
- explain here what to do to test this (or point to unit tests)

todo:
 - [x] add endpoint to create and fetch triggers
    - [x] update ? ( or just to recreate)
 - [x] test endpoint
 - [x] write js module to fetch the triggers and apply the triggers 
     - [x] create/build lib  
     - [x] use lib in app ( start with ssr-catalogue)
     to solve cors script fetch issue see https://github.com/DavidWells/analytics/blob/master/packages/analytics-plugin-google-analytics/src/browser.js
 - [x] write initial trigger handlers ( initial can be just to site improver),
     - [x] evaluate opensource lib instead of writing it
     - [ ] (skip for v1) add endpoint to handle events 
     - [ ] (skip for v1)add some ui to view events
- [ ] add ui to crud triggers
